### ISSUE TITLE
filenames with invalid characters on Windows

### ISSUE THREAD

**Issue by rainforest1155** [owncloud/notes#86]
Is the filename sanitation handled in the app or in core? 
I ran into an issue pasting a url into a note. My Windows Owncloud 1.6.3 sync client refused to sync that note with this error:
10:23     Notes/http:www.schrankmonster.de20140625raspberrypi-sonos-speakers-support-apple-airplay.txt       ownCloud       File contains invalid characters that can not be synced cross platform.

So it looks like the slashes were filtered out, but the colon remained, which isn't supported on Windows. I noticed the same issue with a question mark. Both should be pretty common, even in non url note content. 

For now, I worked around the issue by adding a new line to keep the urls out of the first line.

I'm using Notes app 0.9 on Owncloud 7.0.4. This is running on Debian 7 with Nginx. My browser is Opera 26 on Windows 7.

PS: Why is it that version 0.8 is still marked as the latest release when there's already 0.9?

**Comment by Raydiation** [comment]
Filename sanitation is handled in core, reopen there

**Comment by ericfont** [comment]
I ran into this same issue and determined that it happens because owncloud ignores syncing of files with , :, ?, *, ”, >, <, | in their filename according to these specific rules:

https://doc.owncloud.org/desktop/1.7/architecture.html#ignored-files

I can't find any issues specific to sanitation in core (maybe I'm not searching correctly).  But what I did discover from searching core issues, as far as I can tell, is that the apparent agreement is to not sync files with unsupported names.  Therefore, shouldn't the responsibility be on the notes app to provide a valid filename?  Either remove the invalid characters from the filename or encode the invalid characters using alternative valid ones.
